File: sdks/python/read_dir/extension/req.py
from typing import List, Dict

from api.base_model import CyanBaseModel

from api.core.answer_req import AnswerReq
from api.core.cyan_req import CyanReq


class ExtensionValidateReq(CyanBaseModel):
    answers: List[AnswerReq]
    deterministic_states: List[Dict[str, str]]
    prev_answers: List[AnswerReq]
    prev_cyan: CyanReq
    prev_extension_answers: Dict[str, List[AnswerReq]]
    prev_extension_cyans: Dict[str, CyanReq]
    validate: str


class ExtensionAnswerReq(CyanBaseModel):
    answers: List[AnswerReq]
    deterministic_states: List[Dict[str, str]]
    prev_answers: List[AnswerReq]
    prev_cyan: CyanReq
    prev_extension_answers: Dict[str, List[AnswerReq]]
    prev_extension_cyans: Dict[str, CyanReq]

File: sdks/node/src/api/extension/req.ts
import type { AnswerReq } from '../core/answer_req.js';
import type { CyanReq } from '../core/cyan_req.js';

interface ExtensionValidateReq {
  answers: AnswerReq[];
  deterministicStates: Record<string, string>[];
  prevAnswers: AnswerReq[];
  prevCyan: CyanReq;
  prevExtensionAnswers: Record<string, AnswerReq[]>;
  prevExtensionCyans: Record<string, CyanReq>;
  validate: string;
}

interface ExtensionAnswerReq {
  answers: AnswerReq[];
  deterministicStates: Record<string, string>[];
  prevAnswers: AnswerReq[];
  prevCyan: CyanReq;
  prevExtensionAnswers: Record<string, AnswerReq[]>;
  prevExtensionCyans: Record<string, CyanReq>;
}

// export all
export type { ExtensionAnswerReq, ExtensionValidateReq };
